packages/server/graphql/mutations/resetMeetingToStage.ts

```diff
@@ -0,0 +1,134 @@
+import {GraphQLID, GraphQLNonNull} from 'graphql'
+import GenericMeetingPhase from '../../database/types/GenericMeetingPhase'
+import GenericMeetingStage from '../../database/types/GenericMeetingStage'
+import getRethink from '../../database/rethinkDriver'
+import createNewMeetingPhases, {primePhases} from './helpers/createNewMeetingPhases'
+import ResetMeetingToStagePayload from '../types/ResetMeetingToStagePayload'
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import publish from '../../utils/publish'
+import {getUserId} from '../../utils/authorization'
+import standardError from '../../utils/standardError'
+import findStageById from 'parabol-client/utils/meetings/findStageById'
+import {NewMeetingPhaseTypeEnum} from 'parabol-client/types/graphql'
+
+const resetMeetingToStage = {
+  type: GraphQLNonNull(ResetMeetingToStagePayload),
+  description: `Reset meeting to a previously completed stage`,
+  args: {
+    meetingId: {
+      type: GraphQLNonNull(GraphQLID)
+    },
+    stageId: {
+      type: GraphQLNonNull(GraphQLID)
+    }
+  },
+  resolve: async (_source, {meetingId, stageId}, {authToken, socketId: mutatorId, dataLoader}) => {
+    const r = await getRethink()
+    const operationId = dataLoader.share()
+    const subOptions = {mutatorId, operationId}
+
+    // AUTH
+    const viewerId = getUserId(authToken)
+    const meeting = await dataLoader.get('newMeetings').load(meetingId)
+    if (!meeting) return standardError(new Error('Meeting not found'), {userId: viewerId})
+    const {defaultFacilitatorUserId, facilitatorUserId, phases, teamId, meetingType} = meeting
+    if (viewerId !== facilitatorUserId) {
+      if (viewerId !== defaultFacilitatorUserId)
+        return standardError(new Error('Not meeting facilitator'), {userId: viewerId})
+      return standardError(new Error('Not meeting facilitator anymore'), {userId: viewerId})
+    }
+
+    // VALIDATION
+    const foundResponse = findStageById(phases, stageId)
+    if (!foundResponse)
+      return standardError(new Error('Meeting stage not found'), {userId: viewerId})
+    const {stage: resetToStage} = foundResponse!
+    if (!resetToStage.isNavigableByFacilitator)
+      return standardError(new Error('Stage has not started'), {userId: viewerId})
+    if (!resetToStage.isComplete)
+      return standardError(new Error('Stage has not finished'), {userId: viewerId})
+    if (resetToStage.phaseType !== NewMeetingPhaseTypeEnum.group)
+      return standardError(new Error('Resetting to this stage type is not supported'), {
+        userId: viewerId
+      })
+    if (meeting.endedAt)
+      return standardError(new Error('The meeting has already ended'), {userId: viewerId})
+
+    // RESOLUTION
+    const meetingCount = await r
```

Comment:
you can grab this direct from the meeting & reduce a hit to the DB

Reply:
ah wasn't aware we stored it on the meeting, thanks!